corehq/apps/app_manager/static/app_manager/js/releases/releases.js

```diff
@@ -259,6 +259,14 @@ hqDefine('app_manager/js/releases/releases', function () {
             }
             return hqImport("hqwebapp/js/initial_page_data").reverse.apply(null, arguments);
         };
+        self.webAppsUrl = function(idObservable) {
+            var url = hqImport("hqwebapp/js/initial_page_data").reverse("formplayer_main"),
+                data = {
+                    appId: ko.utils.unwrapObservable(idObservable),
+                };
+
+            return url + '#' + JSON.stringify(data);
```

Comment:
is stringify the result we want here? i think it would append something like `'{"appId":"___"}'` to the url unless i'm missing something

Reply:
Yeah, that's what it does...web apps URLs are weird, like: [https://www.commcarehq.org/a/jennytraining/cloudcare/apps/v2/#{"appId"%3A"ab7527be1f854fddafa5e481215989f5"%2C"steps"%3A[0]%2C"page"%3Anull%2C"search"%3Anull}](https://www.commcarehq.org/a/jennytraining/cloudcare/apps/v2/#{"appId"%3A"ab7527be1f854fddafa5e481215989f5"%2C"steps"%3A[0]%2C"page"%3Anull%2C"search"%3Anull})

Reply:
just realized, i think this might then need url escaping

Reply:
added

Reply:
(it seems to work either way, but might as well be cautious)